src/DataStreams.jl

```diff
@@ -671,7 +690,12 @@ function NamedTuple(sch::Data.Schema{R}, ::Type{S}=Data.Field,
         # for Data.Column or unknown # of rows in Data.Field, we only ever append!, so just allocate empty columns
         rows = ifelse(S == Data.Column, 0, ifelse(!R, 0, sch.rows))
         names = makeunique(Data.header(sch))
-        sink = Base.namedtuple(NamedTuple{names}, (allocate(types[i], rows, reference) for i = 1:length(types))...)
+
+        sink = @static if isdefined(Core, :NamedTuple)
+                Base.namedtuple(NamedTuple{names}, (allocate(types[i], rows, reference) for i = 1:length(types))...)
```

Comment:
For 0.7, this should just be `NamedTuple{names}((allocate(types[i], rows, reference) for i = 1:length(types)))` (I think that should work)